lib/Sema/CSSimplify.cpp

```diff
@@ -3626,18 +3626,37 @@ ConstraintSystem::matchExistentialTypes(Type type1, Type type2,
 
           if (last.is<LocatorPathElt::ApplyArgToParam>()) {
             auto proto = protoDecl->getDeclaredInterfaceType();
-            auto *fix = AllowArgumentMismatch::create(
-                *this, type1, proto, getConstraintLocator(anchor, path));
-
             // Impact is 2 here because there are two failures
             // 1 - missing conformance and 2 - incorrect argument type.
             //
             // This would make sure that arguments with incorrect
             // conformances are not prioritized over general argument
             // mismatches.
+            if (type1->isOptional()) {
+              auto unwrappedType = type1->lookThroughAllOptionalTypes();
+              auto result = simplifyConformsToConstraint(
+                  unwrappedType, type2, ConstraintKind::ConformsTo,
+                  getConstraintLocator(locator),
+                  TypeMatchFlags::TMF_ApplyingFix);
```

Comment:
I believe the reason you are seeing the extra diagnostic is this is not the correct way to call `simplifyConformsToConstraint`. You can try it like this:
```suggestion
              auto result = simplifyConformsToConstraint(unwrappedType, protoDecl, kind, locator,
                                                         subflags | TMF_ApplyingFix);
```

Reply:
Yes, that is correct.

Reply:
Thank you!
 I think I addressed this!